packages/react-instantsearch-core/src/widgets/InstantSearch.tsx

```diff
@@ -53,13 +52,17 @@ type Props = {
     searchState: SearchState
   ) => void;
   stalledSearchDelay?: number;
-  resultsState: SearchResults | { [indexId: string]: SearchResults };
+  resultsState: ResultsState | { [indexId: string]: ResultsState };
 };
 
 type State = {
   contextValue: InstantSearchContext;
 };
 
+function isControlled(props: Props) {
```

Comment:
TS complained about "use before declare", but not sure why it only now complained. It's not that big of a deal though.